tests/mocks/nrf_rpc/mock_nrf_rpc_transport.c

```diff
@@ -85,9 +86,13 @@ static int send(const struct nrf_rpc_tr *transport, const uint8_t *data, size_t
 	expected = &ctx->expected[ctx->cur_expected];
 	response = &ctx->response[ctx->cur_expected];
 	++ctx->cur_expected;
+	log_payload("Expected nRF RPC packet", expected->data, expected->len);
 
-	zassert_equal(expected->len, length, "Unexpected nRF RPC packet sent");
```

Comment:
What is the difference. Seems the previous implementation does the same, but in short form.

Reply:
Zassert interrupts the flow of the test by that k_free is not being called. And the expectations in most of the tests also carry between tests.